src/libbson/src/bson/bson-json.c

```diff
@@ -967,6 +1034,8 @@ _bson_json_read_string (bson_json_reader_t *reader, /* IN */
       case BSON_JSON_LF_BINARY:
       case BSON_JSON_LF_TYPE:
          bson->bson_type_data.binary.is_legacy = true;
+         /* Break omitted intentionally */
```

Comment:
I assumed that we don't need to set `is_legacy` here, but I'm not very familiar with what it's supposed to do. Please advise if I should change this.